### ISSUE TITLE
Desktop systems need SDL_StartTextInput by default...

### ISSUE THREAD

**Issue by icculus** [libsdl-org/sdl2-compat#183]
In SDL2, SDL_StartTextInput is running by default on "desktop" systems (basically things that aren't phones in this case?).

Right now, SDL_TEXTINPUT events aren't delivered to SDL2 apps on these platforms out of the gate, since SDL3 defaults text input to off for everyone.

We need to decide if this is a platform that should have had it on (or maybe its easier to ifndef for iOS and Android?), and call SDL3_StartTextInput() at the end of SDL_CreateWindow (or maybe at the end of the first created window...? I'm not sure.)

**Comment by slouken** [comment]
Yes, we should replicate the logic that SDL2 had, after creating a window.

**Comment by slouken** [comment]
FYI, this reintroduces issues on Steam Deck popping up the on-screen keyboard. We may need to set the hint to prevent enabling the screen keyboard on platforms where we do this.
